engine/access/ingestion2/execution_receipts_consumer.go

```diff
@@ -0,0 +1,133 @@
+package ingestion2
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/common/fifoqueue"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/storage"
+)
+
+const defaultQueueCapacity = 10_000
+
+type ExecutionReceiptConsumer struct {
+	log                      zerolog.Logger
+	collectionExecutedMetric module.CollectionExecutedMetric
+
+	messageHandler       *engine.MessageHandler
+	receiptsMessageQueue engine.MessageStore
+	notifier             engine.Notifier
+	receipts             storage.ExecutionReceipts
+
+	errorMessageRequester ErrorMessageRequester
+}
+
+func NewExecutionReceiptConsumer(
+	log zerolog.Logger,
+	collectionExecutedMetric module.CollectionExecutedMetric,
+	receipts storage.ExecutionReceipts,
+	errorMessageRequester ErrorMessageRequester,
+) (*ExecutionReceiptConsumer, error) {
+	receiptsRawQueue, err := fifoqueue.NewFifoQueue(defaultQueueCapacity)
+	if err != nil {
+		return nil, fmt.Errorf("could not create execution receipts queue: %w", err)
+	}
+
+	receiptsMessageQueue := &engine.FifoMessageStore{FifoQueue: receiptsRawQueue}
+	messageHandler := engine.NewMessageHandler(
+		log,
+		engine.NewNotifier(),
+		engine.Pattern{
+			Match: func(msg *engine.Message) bool {
+				_, ok := msg.Payload.(*flow.ExecutionReceipt)
+				return ok
+			},
+			Store: receiptsMessageQueue,
+		},
+	)
+
+	return &ExecutionReceiptConsumer{
+		log:                      log,
+		collectionExecutedMetric: collectionExecutedMetric,
+		receipts:                 receipts,
+		messageHandler:           messageHandler,
+		notifier:                 engine.NewNotifier(),
+		errorMessageRequester:    errorMessageRequester,
+	}, nil
+}
+
+// Notify processes a new execution receipt payload and notifies the consumer to begin processing queued messages.
+//
+// No errors are expected during normal operations.
+func (c *ExecutionReceiptConsumer) Notify(originID flow.Identifier, payload interface{}) error {
+	err := c.messageHandler.Process(originID, payload)
+	c.notifier.Notify()
+	return err
+}
+
+// StartConsuming starts the execution receipt processing loop. It waits for notifications
+// and processes available receipts until the context is cancelled or an irrecoverable error occurs.
+func (c *ExecutionReceiptConsumer) StartConsuming(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-c.notifier.Channel():
```

Comment:
I like that you switched this to use a message processor.

we can actually use the notifier channel managed by the `messageHandler` here instead of maintaining one within the consumer.
https://github.com/onflow/flow-go/blob/8c30c496cf7b6e3fadad97328697a42f633a4853/engine/enqueue.go#L104-L106